examples_tests/61.BlitFilterTest/normalization/descriptors.glsl

```diff
@@ -0,0 +1,55 @@
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_DESCRIPTORS_INCLUDED_
+#define _NBL_GLSL_BLIT_NORMALIZATION_DESCRIPTORS_INCLUDED_
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_DESCRIPTOR_SET_DEFINED_
+#define _NBL_GLSL_BLIT_NORMALIZATION_DESCRIPTOR_SET_DEFINED_ 0
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_
+#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_ nbl_glsl_blit_normalization_inOutImage
+
+	#ifndef _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_BINDING_DEFINED_
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_BINDING_DEFINED_ 0
+	#endif
+
+	#ifndef _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_
+		#error _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ must be defined to any of (u)image1D/(u)image2D/(u)image3D
+	#endif
+
+	#ifndef _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_FORMAT_
+		#error _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_FORMAT_ must be defined to a valid storage image format
+	#endif
+
+	layout(set = _NBL_GLSL_BLIT_NORMALIZATION_DESCRIPTOR_SET_DEFINED_, binding = _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_BINDING_DEFINED_, _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_FORMAT_) uniform _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_;
```

Comment:
this is wrong, if you need to normalize, you need to blit to an intermediate image with equal channel count but higher precision than the desired output.

So you need two descriptors.

However you can take advantage of this and make your life easier by using only hardware storable and sample-able formats for the intermediate result (so basically whatever has equal or greater number of channel and equal or greater precision in each channel)
This is makes life easier as fixed point formats can only be upgraded to fixed point formats, and floating points only to floating points (because distribution of error cannot match), so less choices to consider.

Reply:
Resolved.